lib/OpenQA/Worker.pm

```diff
@@ -750,18 +751,25 @@ sub _handle_job_status_changed ($self, $job, $event_data) {
     }
 }
 
-sub _load_avg ($field = 2) {
-    my $value = eval { (split(' ', path($ENV{OPENQA_LOAD_AVG_FILE} // '/proc/loadavg')->slurp))[$field] };
+sub _load_avg () {
+    my @load = eval { split(' ', path($ENV{OPENQA_LOAD_AVG_FILE} // '/proc/loadavg')->slurp) };
+    splice @load, 3;    # remove non-load numbers
     log_warning "Unable to determine average load: $@" if $@;
-    return looks_like_number($value) ? $value : undef;
-}
-
-sub _check_system_utilization ($self) {
-    my $settings = $self->settings->global_settings;
-    return undef unless my $threshold = $settings->{CRITICAL_LOAD_AVG_THRESHOLD};
-    my $load_avg = _load_avg;
-    return "The average load $load_avg is exceeding the configured threshold of $threshold."
-      if defined $load_avg && $load_avg >= $threshold;
+    return [] unless all { looks_like_number $_ } @load;
+    return \@load;
+}
+
+sub _check_system_utilization (
+    $self,
+    $threshold = $self->settings->global_settings->{CRITICAL_LOAD_AVG_THRESHOLD},
+    $load = _load_avg())
+{
+    return undef unless $threshold;
+    return undef unless @$load;
```

Comment:
I'd combine these conditions and also check that load has at least 3 elements because the subsequent code relies on that:
```suggestion
    return undef unless $threshold && @$load >= 3;
```

Reply:
Do we need to check if load is defined before checking it's length to prevent perl warnings? Like

```suggestion
    return undef unless $threshold && $load && @$load >= 3;
```

Reply:
No, you always return an array ref